syncapi/syncapi.go

```diff
@@ -50,57 +53,53 @@ func AddPublicRoutes(
 		logrus.WithError(err).Panicf("failed to connect to sync db")
 	}
 
-	pos, err := syncDB.SyncPosition(context.Background())
-	if err != nil {
-		logrus.WithError(err).Panicf("failed to get sync position")
-	}
-
-	notifier := sync.NewNotifier(pos)
-	err = notifier.Load(context.Background(), syncDB)
-	if err != nil {
-		logrus.WithError(err).Panicf("failed to start notifier")
+	eduCache := cache.New()
+	streams := streams.NewSyncStreamProviders(syncDB, userAPI, rsAPI, keyAPI, eduCache)
+	notifier := notifier.NewNotifier(streams.Latest(context.Background()))
+	if err = notifier.Load(context.Background(), syncDB); err != nil {
+		logrus.WithError(err).Panicf("failed to load notifier")
 	}
 
-	requestPool := sync.NewRequestPool(syncDB, cfg, notifier, userAPI, keyAPI, rsAPI)
+	requestPool := sync.NewRequestPool(syncDB, cfg, userAPI, keyAPI, rsAPI, streams, notifier)
 
 	keyChangeConsumer := consumers.NewOutputKeyChangeEventConsumer(
 		cfg.Matrix.ServerName, string(cfg.Matrix.Kafka.TopicFor(config.TopicOutputKeyChangeEvent)),
-		consumer, notifier, keyAPI, rsAPI, syncDB,
+		consumer, keyAPI, rsAPI, syncDB, notifier, streams,
```

Comment:
It would be nicer if we just passed in the stream required by `KeyChangeEventConsumer`, etc.

Reply:
Done for all but the roomserver consumer, which currently uses multiple streams.

Reply:
Now done it for the roomserver consumer too, just passed in more than one.